calico_containers/calicoctl.py

```diff
@@ -356,6 +356,39 @@ def node(ip, node_image, log_dir, ip6="", as_num=None):
     # Print warnings for any known system issues before continuing
     checksystem(fix=False, quit_if_error=False)
 
+    # Get IP address of host, if none was specified
+    if not ip:
+        ip = get_first_ip()
+        if ip:
+            print "No IP provided. Using detected IP: %s" % ip
+        else:
+            print "Couldn't autodetect a management IP address. Please rerun" \
+                  "command with the --ip=<IP_ADDRESS> flag."
+            sys.exit(1)
+
+    # Warn if this hostname is already in use.
+    # If there's already a calico-node container on this host, they're probably
+    # just re-running node to do an update, so only issue a warning
```

Comment:
You're right. Its the comment that needs fixing, not the behavior, right?

The desired behavior is that - If there is a calico-node on this container, the etcd entry for this hostname must be this host. Right?